packages/mdc-fab/_mixins.scss

```diff
@@ -0,0 +1,147 @@
+// Copyright 2016 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+@import "@material/animation/functions";
+@import "@material/animation/variables";
+@import "@material/elevation/mixins";
+@import "@material/ripple/mixins";
+@import "@material/theme/mixins";
+@import "./variables";
+
+@mixin mdc-fab-selector-disabled($selector) {
+  :disabled #{$selector},
+  #{$selector}:disabled {
+    @content;
+  }
+}
+
+@mixin mdc-fab-base {
+  @include mdc-ripple-base;
+
+  display: inline-flex;
+  position: relative;
+  justify-content: center;
+  width: 56px;
+  height: 56px;
+  padding: 0;
+  transition: box-shadow 280ms $mdc-animation-standard-curve-timing-function;
+  border: none;
+  border-radius: 50%;
+  cursor: pointer;
+  user-select: none;
+  box-sizing: border-box;
+  fill: currentColor;
+  -moz-appearance: none;
+  -webkit-appearance: none;
+  overflow: hidden;
+
+  &:active,
+  &:focus {
+    // TODO(acdvorak): Should this be paired with states and/or ripple? We don't want to disable outline
+    // (an accessibility/usability feature) unless we're confident that there is also a visual indication that the
+    // element has focus. If the client has customized the DOM in some unexpected way, and is certain that another
+    // element will receive focus instead, they can always override this property manually in their CSS.
+    outline: none;
+  }
+
+  &:hover {
+    cursor: pointer;
+  }
+
+  &::-moz-focus-inner {
+    padding: 0;
+    border: 0;
+  }
+
+  // This allows for using SVGs within them to align properly in all browsers.
+  // Can remove once: https://bugzilla.mozilla.org/show_bug.cgi?id=1294515 is resolved.
+
+  // stylelint-disable selector-max-type
+  // postcss-bem-linter: ignore
+  > svg {
+    width: 100%;
+  }
+  // stylelint-enable selector-max-type
+}
+
+//**
+// @param {color|ThemeProp} $fab-theme-config.bg-color
+// @param {color|ThemeProp} $fab-theme-config.fg-color
+// @param {RippleConfig} $fab-theme-config.ripple-config @see mdc-ripple-default-config_ in mdc-ripple/_mixins.scss.
+// @param {color|ThemeProp} $fab-theme-config.tap-highlight-color
+// @param {int} $fab-theme-config.resting-elevation
+// @param {int} $fab-theme-config.active-elevation
+//*
+@mixin mdc-fab-theme($fab-theme-config) {
+  $bg-color: map-get($fab-theme-config, bg-color);
+  $fg-color: map-get($fab-theme-config, fg-color);
+  $ripple-config: map-get($fab-theme-config, ripple-config);
```

Comment:
Based on the state discussion last week we had, the ripple color(ink color) will always be the same as text color, which in this case I think it should be `fg-color`?